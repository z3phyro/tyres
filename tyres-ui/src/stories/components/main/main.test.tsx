import { expect, test } from "vitest";
import { render } from "@solidjs/testing-library";
import Main from "./main";

test("Main renders", () => {
  const { getByRole } = render(() => <Main> </Main>);

  const main = getByRole("main") as HTMLElement;
  expect(main).toBeTruthy();
});

test("Main renders with custom content", () => {
  const { getByRole } = render(() => <Main>1</Main>);

  const main = getByRole("main");
  expect(main.textContent).toBe("1");
});

test("Main snapshot test", () => {
  const { asFragment } = render(() => <Main> </Main>);

  expect(asFragment()).toMatchSnapshot();
});
