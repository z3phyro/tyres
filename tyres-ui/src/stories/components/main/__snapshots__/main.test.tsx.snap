// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main snapshot test 1`] = `"<main class="px-4 pt-16 sm:pt-4 w-full bg-white"> </main>"`;
